packages/modules/viewer-react/src/components/app-ui/EcsqlWidget.tsx

```diff
@@ -0,0 +1,141 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { UiFramework } from "@itwin/appui-react";
+import { QueryRowFormat } from "@itwin/core-common";
+import {
+  Button,
+  ExpandableBlock,
+  LabeledTextarea,
+  Table,
+} from "@itwin/itwinui-react";
+import React, { useState } from "react";
+
+export default function EcsqlWidget() {
+  const [input, setInput] = useState<string>("");
+  const [rows, setRows] = useState<Record<string, string>[]>([]);
+  const [headers, setHeaders] = useState<string[]>([]);
+  const [isloading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string>("");
+
+  const getData = async () => {
+    try {
+      setRows([]);
+      setHeaders([]);
+      setLoading(true);
+      const results = UiFramework.getIModelConnection()!.query(
+        input,
+        undefined,
+        { rowFormat: QueryRowFormat.UseJsPropertyNames }
+      );
+      const rows = [];
```

Comment:
be careful with var names; make sure to look at the scope.
youre reusing the same names in the scope of this function but also at the component level